### DISCUSSION TITLE
keep-intermediates behavior on error

### DISCUSSION THREAD

**Discussion by CraftSpider** [tectonic-typesetting/tectonic#1126] [General]
Currently, it looks like keep-intermediates intentionally skips non-log files when an error occurs. This surprised me, since I was using it specifically because I wanted to see the resulting aux file after an error. Is there a reason this behavior was chosen? I feel keeping aux files is *most* useful when there are unexpected errors in them.

**Comment by kpym** [comment]
I understand that in certain specific cases, examining the `aux` file might offer some utility, though these instances are relatively infrequent. What specific errors are you expecting to uncover within the `aux` file that wouldn't typically be identifiable within the `log` file?

> I feel keeping aux files is **most useful** when there are unexpected errors in them.

Auxiliary files in LaTeX **play a crucial role** in ensuring the seamless compilation (during the seccond pass) and generation of complex documents, **especially when there aren't any apparent errors**. These files contain information that enhances the overall document structure, aiding in the correct placement of references, citations, tables of contents, cross-references, and other elements.

**Reply by CraftSpider** [reply]
I'm coming out this from the perspective of a developer on tectonic itself - I was debugging #1105, which required me to see the intermediate files that were causing the error.

'Most useful' may be an exaggeration caused by my situation, I know what the files are used for but I mostly am interacting with them not as a standard user but as someone who is trying to find out why something went horribly wrong. I tend to look at them because there's a bug in tectonic, which if it was obvious from the logs likely wouldn't need such investigation.

**Reply by kpym** [reply]
@CraftSpider Sorry for the misunderstanding. I agree that if `--keep-intermediates` is provided, the `.aux` should be written to disk even on error (even if it is not complete). Perhaps you should fill an issue by providing a MWE that demonstrates the problem.

**Comment by pkgw** [comment]
I don't recall there being any principled motivation behind this choice of behavior. And I agree that it would make sense to keep these kinds of files around when errors happen!
